Sprint 7

Kirillov1515 opened the pull request.

Kirillov1515 committed:
Начало работы над спринтом 7

Kirillov1515 committed:
Выполнено задание 7 спринта

So1ovyov commented on MovieQuiz/Presentation/MovieQuizPresenter.swift:
```diff
@@ -0,0 +1,131 @@
+    
+    private weak var viewController: MovieQuizViewControllerProtocol?
+    private var questionFactory: QuestionFactoryProtocol?
+    private let statisticService: StatisticService!
```
Практика force property не безопасна, давай используем опционал.

Kirillov1515 replied:
Удивился, когда увидел force property в лекциях в данном месте, но решил оставить, раз в лекциях так указано. Все исправил и в MovieQuizPresenter и в MovieQuizViewController

So1ovyov commented on MovieQuiz/Presentation/MovieQuizPresenter.swift:
```diff
@@ -0,0 +1,131 @@
+        viewController.showLoadingIndicator()
+    }
+    
+    func isLastQuestion() -> Bool {
```
Кажется необходим модификатор `private`, проверь весь проект на необходимость данного модификатора у свойств и методов, которые не должны торчать наружу.

Kirillov1515 replied:
Согласен, спасибо! Везде, где нужно добавил. Опять же сдал так, потому что в итоговом варианте лекций было реализовано именно так. Видимо стоит отправить фидбек по данному месту в лекциях в поддержку практикума.

So1ovyov replied:
Я просто как ревьюер подсветил этот момент. Вдруг у тебя были идеи сделать так )

Kirillov1515 replied:
Да, конечно, спасибо большое за комментарии!

So1ovyov commented on MovieQuizTests/ArrayTests.swift:
```diff
@@ -0,0 +1,28 @@
+import XCTest
+@testable import MovieQuiz
+
+class ArrayTests: XCTestCase {
```
Давай добавим для всех классов с тестами модификатор `final`

Kirillov1515 replied:
Все добавил

Kirillov1515 committed:
1. Убрал force property
2. Добавил модификатор private ко всем методам, где необходимо
3. Добавил модификатор final для класса с тестами
